### ISSUE TITLE
Created machines do not show up as nodes when running `kubectl get node`

### ISSUE THREAD

**Issue by sflxn** [kubernetes-sigs/cluster-api-provider-vsphere#220]
I create a worker machine.  The machine object shows up when running `kubectl get machines` but does not show up when running `kubectl get nodes`.  This appears to be a recent development as I believe it used to work.

```
apiVersion: "cluster.k8s.io/v1alpha1"
kind: MachineList
items:
- apiVersion: "cluster.k8s.io/v1alpha1"
  kind: Machine
  metadata:
    name: node-1
    labels:
      cluster.k8s.io/cluster-name: test1
  spec:
    providerSpec:
      value:
        apiVersion: "vsphereproviderconfig/v1alpha1"
        kind: "VsphereMachineProviderConfig"
        vsphereMachine: "standard-node"
        machineSpec:
          datacenter: "vicdc"
          datastore: "datastore108"
          resourcePool: "kube-resource-pool"
          networks:
          - networkName: "VM Network"
            ipConfig:
              networkType: dhcp
              ip: ""
              netmask: ""
              gateway: ""
              dns:
          numCPUs: 2
          memoryMB: 2048
          template: "xenial-server-cloudimg-amd64"
          disks:
          - diskLabel: "Hard disk 1"
            diskSizeGB: 20
          preloaded: false
    versions:
      kubelet: 1.11.3
      controlPlane: 1.11.3
    roles:
    - Node
```

**Comment by sflxn** [comment]
Looks like it's trying to bring up the node as a master, the cloud-init-output.log on the node has the following tidbits:

```
[markmaster] Marking the node node-1 as master by adding the label "node-role.kubernetes.io/master=''"
[markmaster] Marking the node node-1 as master by adding the taints [node-role.kubernetes.io/master:NoSchedule]
```

and

```
Your Kubernetes master has initialized successfully!
```

**Comment by sflxn** [comment]
Closing this bug.  Upon further investigation, it appears the two cases I was investigating had explanations for why the node didn't appear.  The one above, where I created a single node listed `controlPlane` in the versions section so it was started up with the master node cloud init.  The second case, where I was starting up a machineset, the machine object appeared, but the k8s node was still starting up.  It just took a little longer for the node to appear because the system was a little slow.
